pkg/plugins/common/kustomize/v1/scaffolds/api.go

```diff
@@ -0,0 +1,96 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaffolds
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/crd"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/crd/patches"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/rbac"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/samples"
+)
+
+var _ plugins.Scaffolder = &apiScaffolder{}
+
+// apiScaffolder contains configuration for generating scaffolding for Go type
+// representing the API and controller that implements the behavior for the API.
+type apiScaffolder struct {
+	config   config.Config
+	resource resource.Resource
+
+	// fs is the filesystem that will be used by the scaffolder
+	fs machinery.Filesystem
+
+	// force indicates whether to scaffold files even if they exist.
+	force bool
+}
+
+// NewAPIScaffolder returns a new Scaffolder for API/controller creation operations
+func NewAPIScaffolder(config config.Config, res resource.Resource, force bool) plugins.Scaffolder {
+	return &apiScaffolder{
+		config:   config,
+		resource: res,
+		force:    force,
+	}
+}
+
+// InjectFS implements cmdutil.Scaffolder
+func (s *apiScaffolder) InjectFS(fs machinery.Filesystem) {
+	s.fs = fs
+}
+
+// Scaffold implements cmdutil.Scaffolder
+func (s *apiScaffolder) Scaffold() error {
+	fmt.Println("Writing kustomize manifests for you to edit...")
+
+	// Initialize the machinery.Scaffold that will write the files to disk
+	scaffold := machinery.NewScaffold(s.fs,
+		machinery.WithConfig(s.config),
+		machinery.WithResource(&s.resource),
+	)
+
+	// Keep track of these values before the update
+	doAPI := s.resource.HasAPI()
+
+	if doAPI {
+
+		if err := scaffold.Execute(
+			&samples.CRDSample{Force: s.force},
+			&rbac.CRDEditorRole{},
+			&rbac.CRDViewerRole{},
+			&patches.EnableWebhookPatch{},
+			&patches.EnableCAInjectionPatch{},
+		); err != nil {
+			return fmt.Errorf("error scaffolding APIs: %v", err)
+		}
+
+		if err := scaffold.Execute(
+			&crd.Kustomization{},
+			&crd.KustomizeConfig{},
+		); err != nil {
+			return fmt.Errorf("error scaffolding kustomization: %v", err)
+		}
```

Comment:
Does it still make sense to keep them two different `Execute` calls with 2 error messages?

Reply:
Probably not.